src/main/java/com/google/api/codegen/discovery/DefaultString.java

```diff
@@ -0,0 +1,150 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.discovery;
+
+import com.google.api.codegen.Inflector;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+/**
+ * Creates default string from path patterns.
+ */
+public class DefaultString {
+
+  private static final String WILDCARD_PATTERN = "[^/]*";
+
+  /**
+   * Returns a default string from `pattern`, or null if the pattern is not supported.
+   */
+  public static String of(String pattern) {
+    // We only care about patterns that look like
+    //  ^projects/[^/]*/subscriptions/[^/]*$
```

Comment:
I see the confusion. We actually accept the pattern if it's alternates between literal and wildcard. Add clarification.